### PR TITLE
Add list of keyboard shortcuts to document settings

### LINKED ISSUE #373: Add keyboard shortcuts tab to document settings

### PR DESCRIPTION
**Pull request by shueja**
Closes #373

Ctrl/Command key is shown depending on the user's detected OS

image
image
image
image

### CODE REVIEW

File: src/components/config/KeyboardShortcutsPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  },
+  File: {
+    "⌘ + N": "New File",
+    "⌘ + O": "Open File",
+    "⌘ + Z": "Undo",
+    "⌘ + Shift + Z, ⌘ + Y": "Redo"
+  },
+  Waypoints: {
+    "1 through 4": "Select waypoint type in navbar",
+    "Shift-1 through Shift-4": "Change type of selected waypoint",
```
Comment (Awesomeplayer165) [inline src/components/config/KeyboardShortcutsPanel.tsx:44]:
```suggestion
    "Shift+1 through Shift+4": "Change type of selected waypoint",
```

File: src/components/config/KeyboardShortcutsPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    isMac = (type == "Darwin");
+})
+// These default to the Mac ⌘ key because it's easier to replace with Ctrl than vice versa
+let shortcuts = {
+  Path: {
+    "G, ⌘ + G": "Generate current path",
+    Space: "Start/stop path playback",
+    Escape: "Clear sidebar and navbar selection"
+  },
+  "Field View": {
```
Comment (Awesomeplayer165) [inline src/components/config/KeyboardShortcutsPanel.tsx:29]: why are some keys with quotes and some without? Are we trying to keep single words without quotes? Is there a purpose between differentiating?

File: src/components/config/KeyboardShortcutsPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    "Right Arrow, Z": "Select next waypoint in path",
+    "Delete/Backspace": "Delete selected waypoint"
+  },
+  "Selected Waypoint Movement": {
+    "Left Mouse Drag on Center": "Move Waypoint",
+    "Left Mouse Drag on Edge": "Rotate Waypoint",
+    E: "Rotates 1/32 turn clockwise",
+    Q: "Rotates 1/32 turn counterclockwise",
+    "Shift + Rotation Key": "Rotate by 1/8 turn instead of 1/32 turn",
+    W: "Translate 0.1 m up (+Y)",
```
Comment (Awesomeplayer165) [inline src/components/config/KeyboardShortcutsPanel.tsx:57]: Doesn't this decrease y value?

File: src/components/config/SettingsModal.tsx [hunk 4]
```diff
@@ -68,17 +69,28 @@ class SettingsModal extends Component<Props, State> {
               <Tab label="Robot Config" />
               <Tab label="Export Config" />
               <Tab label="Betas" />
+              <Tab label="Controls" />
```
Comment (Awesomeplayer165) [inline src/components/config/SettingsModal.tsx:72]: should controls go before betas?

File: src/components/config/SettingsModal.tsx [hunk 5]
```diff
@@ -68,17 +69,28 @@ class SettingsModal extends Component<Props, State> {
...
+              overflowY: "scroll"
+            }}
+          >
             {uiState.settingsTab == 0 && <RobotConfigPanel></RobotConfigPanel>}
             {uiState.settingsTab == 1 && (
               <ExportConfigPanel></ExportConfigPanel>
             )}
             {uiState.settingsTab == 2 && <BetasConfigPanel></BetasConfigPanel>}
+            {uiState.settingsTab == 3 && (
+              <KeyboardShortcutsPanel></KeyboardShortcutsPanel>
```
Comment (Awesomeplayer165) [inline src/components/config/SettingsModal.tsx:92]: idk about the rest of the codebase, but some places you are using `<KeyboardShortcutsPanel />` and some `<KeyboardShortcutsPanel>`

File: src/components/config/KeyboardShortcutsPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+type State = object;
+
+let isMac = false;
+// We don't want to block during the imports stage, especially
+// if type() somehow fails.
+type().then(type=>{
+    isMac = (type == "Darwin");
+})
+// These default to the Mac ⌘ key because it's easier to replace with Ctrl than vice versa
+let shortcuts = {
```
Comment (Awesomeplayer165) [inline src/components/config/KeyboardShortcutsPanel.tsx:23]: all of these should be flipped. I see people looking for what is the generate key to find the shortcut. not, oh i forgot what command G does?

tldr should be descriptions on left and the shortcut on right (flipped).

**Comment by Awesomeplayer165** [comment]
What is this gap doing here?

File: src/components/config/KeyboardShortcutsPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+          fontSize: "2rem",
+          marginTop: `${1 * this.rowGap}px`,
+          marginBottom: `${1 * this.rowGap}px`,
+          marginInline: "auto",
+          width:"max-content"
+        }}
+      >
+        {Object.entries(shortcuts).map((entry) => (
+          <>
+            <Accordion>
```
Comment (Awesomeplayer165) [inline src/components/config/KeyboardShortcutsPanel.tsx:80]: What is this gap doing here?

**Review by Awesomeplayer165 — CHANGES_REQUESTED** [review]

File: src/components/config/KeyboardShortcutsPanel.tsx (hunk 3 above)
Comment (Awesomeplayer165) [reply src/components/config/KeyboardShortcutsPanel.tsx:57]: (Line 57)

Comment (shueja) [reply src/components/config/KeyboardShortcutsPanel.tsx:57]: No

Comment (Awesomeplayer165) [reply src/components/config/KeyboardShortcutsPanel.tsx:57]: it does

Comment (shueja) [reply src/components/config/KeyboardShortcutsPanel.tsx:57]: I don't know what you're seeing but upwards on the screen is increasing Y

File: src/components/config/SettingsModal.tsx (hunk 4 above)
Comment (shueja) [reply src/components/config/SettingsModal.tsx:72]: Sure

File: src/components/config/KeyboardShortcutsPanel.tsx (hunk 2 above)
Comment (shueja) [reply src/components/config/KeyboardShortcutsPanel.tsx:29]: The Prettier formatter unquoted the ones with single words. IDK if it's worth figuring out how to ignore it here

File: src/components/config/KeyboardShortcutsPanel.tsx (hunk 6 above)
Comment (shueja) [reply src/components/config/KeyboardShortcutsPanel.tsx:23]: Done

File: src/components/config/KeyboardShortcutsPanel.tsx (hunk 7 above)
Comment (shueja) [reply src/components/config/KeyboardShortcutsPanel.tsx:80]: Fixed

File: src/components/config/KeyboardShortcutsPanel.tsx (hunk 2 above)
Comment (Awesomeplayer165) [reply src/components/config/KeyboardShortcutsPanel.tsx:29]: then thats fine.
